qds_sdk/clusterv2.py

```diff
@@ -0,0 +1,545 @@
+from qds_sdk.qubole import Qubole
+from qds_sdk.resource import Resource
+from qds_sdk.cloud.cloud import Cloud
+from qds_sdk.engine import Engine
+from qds_sdk import util
+import argparse
+import json
+
+def str2bool(v):
+    return v.lower() in ("yes", "true", "t", "1")
+
+class ClusterCmdLine:
+
+    @staticmethod
+    def parsers(action):
+        argparser = argparse.ArgumentParser(
+            prog="qds.py cluster",
+            description="Cluster Operations for Qubole Data Service.")
+        subparsers = argparser.add_subparsers(title="Cluster operations")
+
+        if action == "create":
+            create = subparsers.add_parser("create", help="Create a new cluster")
+            ClusterCmdLine.create_update_clone_parser(create, action="create")
+            create.set_defaults(func=ClusterV2.create)
+
+        if action == "update":
+            update = subparsers.add_parser("update", help="Update the settings of an existing cluster")
+            ClusterCmdLine.create_update_clone_parser(update, action="update")
+            update.set_defaults(func=ClusterV2.update)
+
+        if action == "clone":
+            clone = subparsers.add_parser("clone", help="Clone a cluster from an existing one")
+            ClusterCmdLine.create_update_clone_parser(clone, action="clone")
+            clone.set_defaults(func=ClusterV2.clone)
+
+        return argparser
+
+    @staticmethod
+    def create_update_clone_parser(subparser, action=None):
+        # cloud config parser
+        cloud = Qubole.cloud_config
```

Comment:
Right now also it will return cached object. I will be calling get_cloud function once during configuration . And after that user can use Qubole.cloud_config everywhere accordingly. If the user want to change it then they can directly call get_cloud by passing cloud_provider.